UtopiaCity/Controllers/Life/WeatherController.cs

```diff
@@ -0,0 +1,52 @@
+﻿using AutoMapper;
+using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using UtopiaCity.Helpers.WeatherReportApi;
+
+namespace UtopiaCity.Controllers.Life
+{
+    public class WeatherController : Controller
+    {
+        private readonly IMapper _mapper;
+        public WeatherController(IMapper mapper)
+        {
+            _mapper = mapper;
+        }
+        [HttpGet("flightreports")]
+        public async Task<List<UtopiaCity.Models.Airport.WeatherReport>> GetFlightWeatherReports()
+        {
+            var data = await WeatherReportApi.Get30DayReportAsync();
+            if (data != null)
+            {
+                return _mapper.Map<List<UtopiaCity.Models.Airport.WeatherReport>>(data);
+            }
+            else
+            {
+                return null;
```

Comment:
if mapping gets null, it doesn't work at all?

Reply:
what should i then do? throw exception message "no data"?

> if mapping gets null, it doesn't work at all?

what should i then do? throw exception message "no data"?

Reply:
maybe return NotFound()?

Reply:
> maybe return NotFound()?

got it